test/collection_interval/collection_interval_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package collection_interval
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/test"
+)
+
+const (
+	// Let the agent run for 2 minute intervals.
+	agentRuntime    = 2 * time.Minute
+	metricName      = "disk_used_percent"
+	periodInSeconds = 60
+	retryCount      = 3
+)
+
+type input struct {
+	lowerBoundInclusive int
+	upperBoundInclusive int
+	dataInput           string
+}
+
+// Sets the collection interval and makes sure we get the numbers of expected metrics
+// With the chance of jitter the sample count of metrics could be lower or higher than expected number of metrics
+// The bounds are to account for jitter
+// Add a shorter force flush interval than collection interval to make sure sample count are closer to correct
+func TestCollectionInterval(t *testing.T) {
+
+	parameters := []input{
+		{dataInput: "resources/default.json", lowerBoundInclusive: 1, upperBoundInclusive: 3},
+		{dataInput: "resources/agent_interval_10s.json", lowerBoundInclusive: 11, upperBoundInclusive: 13},
+		{dataInput: "resources/metric_interval_10s.json", lowerBoundInclusive: 11, upperBoundInclusive: 13},
+		{dataInput: "resources/metric_override_interval_10s.json", lowerBoundInclusive: 11, upperBoundInclusive: 13},
+	}
+
+	for _, parameter := range parameters {
+		t.Run(fmt.Sprintf("resource file location %s number of metrics lower bound %d and upper bound %d",
+			parameter.dataInput, parameter.lowerBoundInclusive, parameter.upperBoundInclusive), func(t *testing.T) {
```

Comment:
nit: I would prefer to have this as test/input name in the `parameters` but I think I see why you keep in this way as you wanna use `dataInput`/`lowerBoundInclusive`/`upperBoundInclusive` to construct the string. Not blocking this.